src/anemoi/training/utils/checkpoint.py

```diff
@@ -91,3 +91,21 @@ def transfer_learning_loading(model: torch.nn.Module, ckpt_path: Path | str) ->
     # Load the filtered st-ate_dict into the model
     model.load_state_dict(state_dict, strict=False)
     return model
+
+
+def freeze_submodule_by_name(module: nn.Module, target_name: str) -> None:
+    """
+    Recursively freezes the parameters of a submodule with the specified name.
+
+    Args:
+        module (nn.Module): The parent module to search in.
+        target_name (str): The name of the submodule to freeze.
```

Comment:
This is the wrong docstring format. Could you please adjust this to the standard used throughout Anemoi (it's the numpy style).